src/test/javascript/spec/component/assessment-dashboard/exam-assessment-buttons.component.spec.ts

```diff
@@ -0,0 +1,238 @@
+                    new HttpResponse({
+                        body: studentExams,
+                        status: 200,
+                    }),
+                );
+            },
+        }),
+        MockProvider(CourseManagementService, {
+            find: () => {
+                return of(
+                    new HttpResponse({
+                        body: course,
+                        status: 200,
+                    }),
+                );
+            },
+        }),
+        MockProvider(AlertService),
+        MockProvider(ArtemisTranslatePipe),
+        {
+            provide: ActivatedRoute,
+            useValue: {
+                params: {
+                    subscribe: (fn: (value: Params) => void) =>
+                        fn({
+                            courseId: 1,
+                        }),
+                },
+                snapshot: {
+                    paramMap: convertToParamMap({
+                        courseId: '1',
+                        examId: '1',
+                    }),
+                },
+            },
+        },
+        { provide: AccountService, useClass: MockAccountService },
+    ];
+
+    beforeEach(() => {
+        course = new Course();
+        course.id = 1;
+
+        studentOne = new User();
+        studentOne.id = 1;
+
+        exam = new Exam();
+        exam.course = course;
+        exam.id = 1;
+        exam.registeredUsers = [studentOne];
+        exam.endDate = dayjs();
+        exam.startDate = exam.endDate.subtract(60, 'seconds');
+
+        studentExamOne = new StudentExam();
+        studentExamOne.exam = exam;
+        studentExamOne.id = 1;
+        studentExamOne.workingTime = 70;
+        studentExamOne.user = studentOne;
+
+        studentExams = [studentExamOne];
+
+        return TestBed.configureTestingModule({
+            imports: [],
+            declarations: [ExamAssessmentButtonsComponent],
+            providers,
+        })
+            .compileComponents()
+            .then(() => {
+                examAssessmentButtonsFixture = TestBed.createComponent(ExamAssessmentButtonsComponent);
+                examAssessmentButtonsComponent = examAssessmentButtonsFixture.componentInstance;
+                examManagementService = TestBed.inject(ExamManagementService);
+            });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should automatically assess modeling and text exercises of unsubmitted student exams', () => {
+        studentExamOne!.workingTime = 10;
+        exam.startDate = dayjs().subtract(200, 'seconds');
+        exam.endDate = dayjs().subtract(100, 'seconds');
+        exam.gracePeriod = 0;
+        course.isAtLeastInstructor = true;
+
+        examAssessmentButtonsFixture.detectChanges();
+        expect(examAssessmentButtonsComponent.isLoading).toEqual(false);
+        expect(examAssessmentButtonsComponent.isExamOver).toEqual(true);
+        expect(course).toBeTruthy();
```

Comment:
Can this ever fail? You're setting something in line 144 so it can't really happen. Same for the other tests below. If I'm wrong, I think this should be replaced with a more meaningful tests (avoid toBeTruthy / Falsy)

Reply:
I have merely copied these tests from their original location in the student-exams.component.spec i didnt make them myself. As this PR has been open for a pretty long time i suggest this can be done in a follow up that is specifically about test quality.